modules/solid_mechanics/src/timeintegrators/ExplicitMixedOrder.C

```diff
@@ -0,0 +1,341 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "Assembly.h"
+#include "ExplicitMixedOrder.h"
+#include "ExplicitTimeIntegrator.h"
+#include "Moose.h"
+#include "MooseError.h"
+#include "MooseTypes.h"
+#include "MooseVariableFieldBase.h"
+#include "NonlinearSystem.h"
+#include "FEProblem.h"
+#include "TimeStepper.h"
+#include "TransientBase.h"
+
+// libMesh includes
+#include "TransientBase.h"
+#include "libmesh/id_types.h"
+#include "libmesh/nonlinear_solver.h"
+#include "libmesh/sparse_matrix.h"
+#include "DirichletBCBase.h"
+#include "libmesh/vector_value.h"
+#include <algorithm>
+#include <iterator>
+#include <utility>
+
+registerMooseObject("SolidMechanicsApp", ExplicitMixedOrder);
+registerMooseObjectRenamed("SolidMechanicsApp",
+                           DirectCentralDifference,
+                           "10/14/2025 00:00",
+                           ExplicitMixedOrder);
+
+InputParameters
+ExplicitMixedOrder::validParams()
+{
+  InputParameters params = ExplicitTimeIntegrator::validParams();
+
+  params.addClassDescription(
+      "Implementation of explicit time integration without invoking any of the nonlinear solver.");
+
+  params.addParam<bool>("use_constant_mass",
+                        false,
+                        "If set to true, will only compute the mass matrix in the first time step, "
+                        "and keep using it throughout the simulation.");
+  params.addParam<TagName>("mass_matrix_tag", "mass", "The tag for the mass matrix");
+
+  params.addParam<std::vector<VariableName>>(
+      "second_order_vars",
+      {},
+      "A subset of variables that require second-order integration (velocity and acceleration) to "
+      "be applied by this time integrator.");
+
+  params.addParam<std::vector<VariableName>>(
+      "first_order_vars",
+      {},
+      "A subset of variables that require first-order integration (velocity only) to be applied by "
+      "this time integrator.");
+
+  // Prevent users from using variables option by accident.
+  params.suppressParameter<std::vector<VariableName>>("variables");
+
+  MooseEnum solve_type("consistent lumped lump_preconditioned", "lumped");
+  params.setParameters("solve_type", solve_type);
+  params.ignoreParameter<MooseEnum>("solve_type");
+  return params;
+}
+
+ExplicitMixedOrder::ExplicitMixedOrder(const InputParameters & parameters)
+  : ExplicitTimeIntegrator(parameters),
+    _constant_mass(getParam<bool>("use_constant_mass")),
+    _mass_matrix(getParam<TagName>("mass_matrix_tag")),
+    _solution_older(_sys.solutionState(2)),
+    _vars_first(declareRestartableData<std::unordered_set<unsigned int>>("first_order_vars")),
```

Comment:
why are these 4 restartable? indexes just get recomputed usually

Reply:
is init() not called on restart / recover?

Reply:
In `TimeIntegrator.C` this is how it's done.